### ISSUE TITLE
Inquiry about some mixtures on DnR

### ISSUE THREAD

**Issue by jvel07** [darius522/dnr-utils#3]
Hi, @darius522 congrats on the repo and corpus, is really useful!

We got a lil question regarding some mixtures of the corpus. We were arbitrarily listening to some test samples, and we found that in some of them when they have music as the background, it's not only instrumented but they also have singing.

Hence, the mixture has, say, 2 'speeches' one from the speech.wav utterance and another one contained in the music.wav track.

Such mixtures are, for instance, 3030 - 11623 - 1882 - 9963.

We intend to experiment with DnR on two separate tasks: speech separation and source separation.
One doubt we have is, having the types of mixtures described above, where an extra kind of 'speech' is present in the mixture (e.g., the one belonging to the background music), isn't this a potential issue for speech separation algorithms?

Likewise, is this a potential issue at all for source separation algos?

We were just curious about this, hope you can provide us with a guide to a better understanding. Thanks in advance! :)

**Comment by darius522** [comment]
Hi @jvel07,

Thank you for your interest in this work! 

This is actually expected. In the context of source separation (or at least how we formulated it in the cocktail fork task) vocals and speech are two distinct components, one belonging to music and the other to speech. I believe that in general, it is a fair distinction to make (otherwise we may offend some vocalists out there!). More practically, imagine a signal in which two people are chatting while some music (with vocals) is playing in the background. Your task is to enhance the chat. Would you want the vocals to be part of it ? I’d say probably not. While our formulation is much more practical (i.e. music with vocals is ubiquitous, and assuming only instrumental music in a real-world setting would be foolish), I agree that it makes the task much more difficult (and may “jeopardize” the predictions quality). 

That being said, vocals in music are usually processed in some unique ways that make them sonically differ from regular speech (i.e. creative effects, compression, reverb, etc.), you could therefore hope that your model would pick up on these cues to make the distinction. In the context of our work and based on some informal listening test, we observed that behavior, for example (while in other cases it failed poorly!).

So in short, yes, it may makes the task more difficult in your case, but for the benefit of (1) approaching it from a more realistic angle, (2) potentially getting some interesting insights as for your model behavior towards vocals and speech.

Hope that helps and please do keep me updated about your findings!

**Comment by jvel07** [comment]
Thanks for your useful reply, @darius522!

I was wondering in this case, what about the mixtures that contain vocals (e.g., an actor singing) and also contain vocals from the background music?
Is the separated voice (vocals of the actor/singer in this case) going to music class or speech in the source separation task?

**Comment by darius522** [comment]
@jvel07  That sounds like a bit of a corner case to me. But once again, it is all about the distribution of your training data, at the end. If your music class data contains many instances of say, A Capella / non-processed musical excerpts, your model may become quite efficient at distinguishing between singing vs. spoken signals. Although, this seems like a very difficult task to generalize upon. I believe that FMA (the music building block used in DnR) does contain A Capella songs, so it may be that a model trained on DnR could already somehow make that distinction.
